### PR TITLE
Fix SSE deadlock

### PR DESCRIPTION
**Pull request by Threated**
- Fix SSE deadlock
- Add SSE test

### CODE REVIEW

File: broker/src/serve_tasks.rs [hunk 1]
```diff
@@ -133,17 +133,12 @@ async fn get_results_for_task_stream(
     }
 
     // There must be a way to just use the stream of stream_results directly but I could not get it to work as the stream returned has a lifetime bound and this one somehow does not
```
Comment (TKussel) [inline broker/src/serve_tasks.rs:135]: This comment is outdated with the change, no?

File: dev/docker-compose.yml
Comment (TKussel) [inline dev/docker-compose.yml:1]: Please remove those comments

**Review by TKussel — CHANGES_REQUESTED** [review]

File: broker/src/serve_tasks.rs (hunk 1 above)
Comment (Threated) [reply broker/src/serve_tasks.rs]: Yes

File: dev/docker-compose.yml
Comment (Threated) [reply dev/docker-compose.yml:1]: Done

**Review by TKussel — APPROVED** [review]

**Pull request merged**